client/components/editor/structure/TesList.vue

```diff
@@ -83,22 +77,18 @@ export default {
     },
     onHiddenElementDrawer() {
       this.isElementDrawerVisible = false;
-      this.insertPosition = this.lastPosition;
+      this.insertPosition = null;
     },
     getContainerClasses({ data: { width } }) {
       let classes = [`col-xs-${width || 12}`];
       if (this.enableAdd) classes.push('insertable');
       return classes;
     },
-    addElement(element) {
-      const type = element.position === this.lastPosition ? 'add' : 'insert';
-      this.$emit(type, element);
-    }
-  },
-  watch: {
-    lastPosition: {
-      handler(val) { this.insertPosition = val; },
-      immediate: true
+    insertElement(element) {
+      let { insertPosition: position, list } = this;
+      if (!Number.isInteger(position)) position = list.length - 1;
```

Comment:
Ummm use `const` :grin:
How we can end up here with `insertPosition` being unset? :thinking:
If that's possible I'm more inclined towards preventing it than ensuring it here...